doc/psyad/user_guide/implementation.rst

```diff
@@ -422,6 +428,15 @@ information on these transformations.
           variables will be detected automatically by PSyAD, see issue
           #1595.
 
+Associativity
+-------------
+
+The transformation from tangent-linear code to adjoint code will fail
```

Comment:
As described in :ref:`Rules`, PSyAD expects all tangent-linear code to be written as a sum of products of inactive and active variables. Therefore if code such as `a(b+c)` is found (where `b` and `c` are active) then it must be transformed into a recognised form. This is achieved by expanding all such expressions as part of the pre-processing phase. In this example, the resulting code is `a*b + a*c` which PSyAD can then take the adjoint of.

Reply:
Much better, thanks.